### DISCUSSION TITLE
Preventing Sumatra From Editing PDF Files?

### DISCUSSION THREAD

**Discussion by AnxiousPleaseHelp** [sumatrapdfreader/sumatrapdf#3158] [General]
Morning,

I just want to know how I can make it impossible for Sumatra to edit PDF files? I want to store PDF files for the long term. 

I just want Sumatra to be a "Read only" PDF reader.

How can this be done?

I hope someone can help me!

**Comment by tactilis** [comment]
> I want to store PDF files for the long term

Select a file (or multiple files) in Windows File Explorer.
Right click and select Properties.
Set the Read-only attribute.
Now it's not possible to accidentally overwrite the file.

![image](https://user-images.githubusercontent.com/4060092/205451115-3851c958-c9e4-4d34-8e7a-adde2e9bf491.png)
 
![image](https://user-images.githubusercontent.com/4060092/205451455-13a7b99d-170e-40df-a134-c06eaf85594d.png)

**Reply by GitHubRulesOK** [reply]
@AnxiousPleaseHelp 

Note this is the same as for ANY computer file i.e. zip e.g. comics images epub doc(x) since ALL files are binary and ALL binary can be edited such as Acrobat.exe or SumatraPDF.exe itself (I will sometimes change tab colours by patching the exe !!) There is nothing special about PDFs they are just as editable as any file.

**Comment by AnxiousPleaseHelp** [comment]
Thanks for your input!

I’m trying to do it now.

Are there any other ways for make it so that .PDF files can’t be changed or accidentally edited?

**Reply by GitHubRulesOK** [reply]
see @tactilis correct answer the only way to stop a computer file from being modified is to burn a ROM (ReadOnlyMemory) device. However that does not stop it being destroyed by sunlight x-ray or intergalactic highways Only Many RE-generations of back-ups are archival (I should know having had to try to recover graveyard data when RAM raided in a ROM War Graves device, luckily it had only been twisted a bit of each byte.) So NO you can't even use that Since I had to edit the files to recover them in order to EDIT & resave.

**Comment by tactilis** [comment]
@AnxiousPleaseHelp 
> Are there any other ways for make it so that .PDF files can’t be changed or accidentally edited?

I don't understand why are you asking for other ways.

You have been shown the simplest and the correct way; by making the PDF file read-only. Doing this makes it impossible for SumatraPDF or any other application to modify the file.

You said "I want to store PDF files for the long term", i.e. without them being edited. The above is the solution.

**Comment by flywire** [comment]
Nothing wrong with OPs request for a RO (read-only) option. Could easily be used on a command line or saved to settings.

As a workaround, OP could start SumatraPDF with a script file appropriate to their OS which would set file to RO then load SumatraPDF and the file. Could even be made fancy to restore read-write (RW) files on exit. this wouldn't help if loading file from within SumatraPDF.

**Reply by GitHubRulesOK** [reply]
@flywire 
thanks for suggestion
@AnxiousPleaseHelp 
you could also use SumatraPDF sumatrapdfrestrict.ini file to do it automatically but as discussed in an issue https://github.com/sumatrapdfreader/sumatrapdf/issues/2846 it may not block all cases where user forces circumvention it can block other desired behaviour so you should use it with caution

**Comment by AnxiousPleaseHelp** [comment]
For anyone that needs help with this: 

**I kept researching this and the best solution I found was to burn Sumatra PDF files to DVD-R discs.**

**Reply by GitHubRulesOK** [reply]
just beware I used to support archival systems and KNOW read ONLY disks can degrade (Some faster than others) so revisit and copy afresh before they self destruct. (Grandfather copy must match Grandchild copy and father be the arbiter to confirm both are mirror images)

**Reply by AnxiousPleaseHelp** [reply]
> just beware I used to support archival systems and KNOW read ONLY disks can degrade (Some faster than others) so revisit and copy afresh before they self destruct. (Grandfather copy must match Grandchild copy and father be the arbiter to confirm both are mirror images)

Wow, thanks!

**Reply by GitHubRulesOK** [reply]
the brand manufacturer should be able to state expected lifespan and the conditions to meet that as clearly the plastics may last 100 years but the coating may not last so long even a few years in sunlight can affect the lifespan Copilot (can be wrong says  WORST CASE

>1. Cheap organic‑dye DVD‑R (the majority sold today)
6 months to 2 years  
Yes, really.
Some batches fail within months if:
stored in sunlight
stored in warm rooms
stored in humid environments
written at maximum speed
manufactured by low‑tier brands (CMC, Ritek, Princo, etc.)
These discs use cyanine or azo dyes, which degrade quickly.

Best case and specialist
>M‑DISC DVD‑R (the only truly archival optical media)
30–100+ years (lab‑tested)  
M‑DISC uses a stone‑like inorganic data layer, not dye.
It is the only optical medium that survives:
heat
humidity
UV
time
But it requires an M‑DISC‑capable burner.
